### PR TITLE
inherit from experimental 'mbed' multi-toolchain target

### PR DESCRIPTION
**Pull request by autopulated**
I'll publish an experimental version of the base `mbed` target that this depends on soon, but works in my local testing with gcc.

related: https://github.com/armmbed/target-mbed

Creating this for discussion!

@bogdanm @bremoran @niklas-arm

### CODE REVIEW

**Comment by bremoran** [comment]
The premise looks sound. What is the expected behaviour of a leaf target in response to this PR?

**Comment by autopulated** [comment]
The first step I think is to do this for both gcc and armcc, while making sure nothing breaks.

Then we'll need to move some things around, and fix any corner cases that do break (coverage things will need to move into the base target).

Then we should be able to add new versions of the intermediate targets (e.g. kinetis-k64-xxx) inheriting from this, instead of the gcc/armcc ones (we could also merge/simplify the existing ones).

Then finally, leaf targets can merge and inherit from the new intermediate targets.

Basically, it should all be non-breaking at every step.

**Comment by bogdanm** [comment]
This looks very good and personally I think it's an extremely good idea. +1.

**Comment by bremoran** [comment]
+1

**Comment by 0xc0170** [comment]
:+1:

**Comment by hugovincent** [comment]
+1

**Comment by niklarm** [comment]
:+1: This will simplify some things for me :-)
